src/main/java/seedu/address/Tasks/TaskList.java

```diff
@@ -0,0 +1,115 @@
+package seedu.address.Tasks;
+
+import seedu.address.commons.exceptions.TaskException;
+import java.util.ArrayList;
+
+/**
+ * The TaskList class represents a list of tasks for the Addressbook program.
+ * It provides methods to manage and manipulate the list of tasks.
+ */
+public class TaskList {
+    ArrayList<Task> taskList;
+
+    /**
+     * Constructs a new TaskList object with an empty list of tasks.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a new TaskList object with the given stored list of tasks.
+     * @param taskList an ArrayList of Task objects representing the list of tasks
+     */
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Deletes a task from the list of tasks.
+     * @param TaskNumber the index of the task to delete (1-based indexing)
+     * @throws TaskException if the given index is out of bounds
+     */
+    public String deleteTask(int TaskNumber) throws TaskException {
```

Comment:
Can return task instead? Then maybe we can standardise what is shown to the GUI through a UI class